cosmos/genesis.go

```diff
@@ -0,0 +1,165 @@
+package cosmos
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/genaccounts"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/privval"
+	"github.com/tendermint/tendermint/types"
+)
+
+// GenesisValidator holds the info of a specific validator to use to generate a genesis.
+type GenesisValidator struct {
+	Name      string
+	Address   sdktypes.AccAddress
+	Password  string
+	Mnemonic  string
+	ValPubKey crypto.PubKey
+	NodeID    p2p.ID
+}
+
+// NewGenesisValidator creates a new validator with an cosmos account, validator and node identity.
+func NewGenesisValidator(kb *Keybase, name, password, privValidatorKeyFile, privValidatorStateFile, nodeKeyFile string) (GenesisValidator, error) {
+	// create validator account
+	mnemonic, err := kb.NewMnemonic()
+	if err != nil {
+		return GenesisValidator{}, err
+	}
+	acc, err := kb.CreateAccount(name, mnemonic, "", password, 0, 0)
+	if err != nil {
+		return GenesisValidator{}, err
+	}
+	val := privval.LoadOrGenFilePV(privValidatorKeyFile, privValidatorStateFile)
+	nodeKey, err := p2p.LoadOrGenNodeKey(nodeKeyFile)
+	if err != nil {
+		return GenesisValidator{}, err
+	}
+	return GenesisValidator{
+		Name:      name,
+		Password:  password,
+		Mnemonic:  mnemonic,
+		Address:   acc.GetAddress(),
+		ValPubKey: val.GetPubKey(),
+		NodeID:    nodeKey.ID(),
+	}, nil
+}
+
+// Map exports the info of GenesisValidator in a nice way for logrus.
+func (validator GenesisValidator) Map() map[string]interface{} {
+	return map[string]interface{}{
+		"name":     validator.Name,
+		"address":  validator.Address,
+		"password": validator.Password,
+		"mnemonic": validator.Mnemonic,
+		"nodeID":   validator.NodeID,
+		"peer":     fmt.Sprintf("%s@%s:26656", validator.NodeID, validator.Name),
+	}
```

Comment:
Why there are all those fields in map, while only `peer` is used?

Reply:
It's for logging when using `logrus.WithFields`. How could I do it nicely?

Reply:
I would like to implement a similar function to `String()` but for logrus that can return this `map[string]interface{}`

Reply:
Either log a struct as json or use `WithFields`  func

```
  log.WithFields(log.Fields{
	"name":     validator.Name,
		"address":  validator.Address,
		"password": validator.Password,
		"mnemonic": validator.Mnemonic,
		"nodeID":   validator.NodeID,
		"peer":     fmt.Sprintf("%s@%s:26656", validator.NodeID, validator.Name),
  }).Info("A walrus appears")
```

Instead of creating a dedicated method for return `map`